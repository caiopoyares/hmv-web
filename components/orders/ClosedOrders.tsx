import {
  Box,
  Heading,
  LinkBox,
  LinkOverlay,
  Tag,
  Text,
} from "@chakra-ui/react";
import { useRouter } from "next/router";

interface Props {
  orders: any[];
}

export const ClosedOrders = ({ orders: allOrders }: Props) => {
  const router = useRouter();
  const closedOrders = allOrders.filter((order) => order.status === "complete");

  const onOrderClick = (orderId: string) => {
    router.push(`/orders/${orderId}/closed`);
  };

  return (
    <Box display="flex" flexDir={["column", "row"]} mt={4}>
      {closedOrders.length < 1 ? (
        <div>
          <Text color="gray.500">Nenhuma ficha encerrada.</Text>
        </div>
      ) : (
        <div>
          {closedOrders.map((order: any) => (
            <LinkBox
              key={order.id}
              as="article"
              maxW="sm"
              p="5"
              borderWidth="1px"
              rounded="md"
              w={["100%", 300]}
              marginRight={[0, 2]}
              marginBottom={[4, 0]}
              onClick={() => onOrderClick(order.id)}
              cursor="pointer"
            >
              <Heading size="md">
                <LinkOverlay>{`${order.user.firstName} ${order.user.lastName}`}</LinkOverlay>
              </Heading>
              <Text color="gray.500">ID do paciente: {order.user.id}</Text>
              <Text mb={4} color="gray.500">
                CPF do paciente: {order.user.cpf}
              </Text>

              <Text mt={4} fontWeight="bold">
                Ficha nยบ {order.id}
              </Text>
              <Text mb={4} color="gray.500">
                Criado em {order.arrivalDate}
              </Text>
              <Tag
                size="md"
                variant="solid"
                bgColor="teal"
                textTransform="uppercase"
              >
                {order.reason}
              </Tag>
            </LinkBox>
          ))}
        </div>
      )}
    </Box>
  );
};
