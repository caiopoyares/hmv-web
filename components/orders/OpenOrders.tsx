import {
  Box,
  Heading,
  HStack,
  LinkBox,
  LinkOverlay,
  Tag,
  Text,
} from "@chakra-ui/react";
import { useRouter } from "next/router";

interface Props {
  orders: any[];
}

export const OpenOrders = ({ orders }: Props) => {
  const router = useRouter();

  const onOrderClick = (orderId: string) => {
    router.push(`/orders/${orderId}/open`);
  };

  return (
    <Box display="flex" flexDir={["column", "row"]} mt={4}>
      {orders.map((order: any) => (
        <LinkBox
          key={order.id}
          as="article"
          maxW="sm"
          p="5"
          borderWidth="1px"
          rounded="md"
          w={["100%", 300]}
          marginRight={[0, 2]}
          marginBottom={[4, 0]}
          onClick={() => onOrderClick(order.id)}
          cursor="pointer"
        >
          <Heading size="md">
            <LinkOverlay>{order.nome}</LinkOverlay>
          </Heading>
          <Text mb={4}>ID do paciente: {order.pacienteId}</Text>
          <Text textTransform="uppercase" fontWeight="bold">
            Ficha nยบ {order.id}
          </Text>
          <Text mb={4} color="gray.500">
            Criado em {order.date}
          </Text>
          <HStack>
            {order.assuntos.map((assunto: any, i: number) => (
              <Tag key={i} size="md" variant="solid" bgColor="teal">
                {assunto}
              </Tag>
            ))}
          </HStack>
        </LinkBox>
      ))}
    </Box>
  );
};
