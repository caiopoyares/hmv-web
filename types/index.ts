export interface Order {
  id: string;
  user: User;
  arrivalDate: Date;
  reason: string;
  description: string;
  status: string;
  suggestions?: string;
  hospital: Hospital;
  doctor?: Doctor;
}

export interface User {
  id: string;
  firstName: string;
  lastName: string;
  cpf: string;
  age: number;
}

export interface Doctor {
  id: string;
  firstName: string;
  lastName: string;
  crm: number;
}

export interface Hospital {
  id: string;
  name: string;
  address: string;
}
