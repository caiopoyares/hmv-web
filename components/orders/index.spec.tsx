import { OpenOrders } from "./OpenOrders";
import { render } from "@testing-library/react";
import { ClosedOrders } from "./ClosedOrders";
import { Order } from "../../types";

const mockedOpenOrders = [
  {
    id: 4,
    user: {
      id: 1,
      firstName: "john",
      lastName: "doe",
      cpf: 42013234276,
    },
    hospital: {
      name: "Mocked hospital",
    },
    reason: "anything",
    arrivalDate: new Date("14/02/2022"),
  } as unknown as Order,
] as Order[];

describe("Orders", () => {
  it("should render empty open orders", () => {
    const { container } = render(<OpenOrders orders={[]} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render open orders", () => {
    const { container } = render(<OpenOrders orders={mockedOpenOrders} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render empty closed orders", () => {
    const { container } = render(<ClosedOrders orders={[]} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render open orders", () => {
    const { container } = render(<ClosedOrders orders={mockedOpenOrders} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

export {};
