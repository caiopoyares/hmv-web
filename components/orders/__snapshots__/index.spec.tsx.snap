// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should render empty closed orders 1`] = `
<div
  class="css-1necnyg"
>
  <p
    class="chakra-text css-ax9bdp"
  >
    Nenhuma ficha encerrada.
  </p>
</div>
`;

exports[`Orders should render empty open orders 1`] = `
<div
  class="css-cjiru7"
>
  <p
    class="chakra-text css-ax9bdp"
  >
    Nenhuma ficha em andamento.
  </p>
</div>
`;

exports[`Orders should render open orders 1`] = `
<div
  class="css-cjiru7"
>
  <p
    class="chakra-text css-ax9bdp"
  >
    Nenhuma ficha em andamento.
  </p>
</div>
`;

exports[`Orders should render open orders 2`] = `
<div
  class="css-1necnyg"
>
  <p
    class="chakra-text css-ax9bdp"
  >
    Nenhuma ficha encerrada.
  </p>
</div>
`;
